hazelcast-client-legacy/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java

```diff
@@ -148,6 +154,17 @@ private void registerInvocation(ClientInvocation clientInvocation) {
         }
     }
 
+    private void checkMaxAllowedReached() {
+        if (maxAllowedConcurrentInvocations == Integer.MAX_VALUE) {
+            return;
+        }
+        long currentConcurrentInvocations = callIdIncrementer.get() - tailCallId.get();
+        if (maxAllowedConcurrentInvocations <= currentConcurrentInvocations) {
+            throw new HazelcastOverloadException("Max allowed concurrent invocations is reached : "
```

Comment:
I think, we should increase `tailCallId` even when there is exception. Otherwise, there will be inconsistency between `callIdIncrementer` and `tailCallId`